language-design.md

```diff
@@ -17,7 +17,24 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+>Names matter. Strive for intelligibility, consistency, and
+>symmetry. Every API is a little language, and people must learn to
+>read and write it. If you get an API right, code will read like
+>prose. [Bloch, 2006]
 
+Both Alex and Zoab felt like this point resonated with them. Especially when reading someone else's code, Zoab has observed that the quality of names affects the ease with which he can understand the code - variable names, method names, etc. that correspond to the semantics of the construct allow you to get an idea of how the construct works without having to closely read the implementation. Alex says "Every name is like a gateway. If you understand the name, you can stop reading. If you don't understand, you have to then go read all of the implementation behind that name."
+
+They both recognize that choosing good names can be a struggle, especially when dealing with abstract or complex constructs, but they both think it's worth putting in the effort. There will always be people who misunderstand your naming convention, but if you put effort into choosing names, you can minimize that set. 
+
+> This leads me to claim that, from now on, a main goal in designing a language should be to plan for growth. [Steele, 1998]
+
+Alex and Zoab don't necessarily agree with this statement. They understand that the person making this statement is a general purpose programming language designer, and they can understand the validity of this claim for general purpose programming languages. However, it's not obvious to them that for DSLs in particular, you'd want to design them with a plan for growth. Consider a language that has already expanded to fill its domain, where its domain is likely to remain static. In this case it wouldn't necessarily be a good idea to plan for growth. An example of this kind of language is Picobot - if there's a trade off between making Picobot easier to use or making it easier to grow, we should choose to make it easy to use because we don't expect to have to grow it. 
+
+> We need to put tools for lanugage growth in the hands of the users. [Steele, 1998]
+
+What does it mean to put the tools for growth into the hands of the users? Alex thinks that means that there should be a way for one users changes to the language to be accepted by the rest of the community. Zoab thinks that one problem with this is that the language could easily grow to be too large because, as we already know, we can't ever take anything out of a language while we're free to add stuff in, so if we make adding stuff easy the growth of the language could quickly spiral out of control. Alex thinks that these issues would still be faced by a central language writing body, and by allowing for any user to contribute we'd have to tackle these issues with a review process. Zoab thinks in practice, even with a review process, bad code and bad style can slip through and become part of the project as a whole. 
```

Comment:
We think that Alex is mostly right about the answer to this question: this means that the users of the language should be able to add to the language standard library or equivalent. Recall that Steele wasn't against large languages _per se_; he thought that a large language had to start as a small one to be adopted, but that large ones were more usable.